openvdb/points/PointMask.h

```diff
@@ -56,33 +56,31 @@ namespace points {
 
 
 /// @brief Extract a Mask Grid from a Point Data Grid
-/// @param grid             the PointDataGrid to extract the mask from.
-/// @param includeGroups    a vector of VDB Points groups to be included (default is all).
-/// @param excludeGroups    a vector of VDB Points groups to be excluded (default is none).
+/// @param grid         the PointDataGrid to extract the mask from.
+/// @param filter       an optional index filter
 /// @note this method is only available for Bool Grids and Mask Grids
 template <typename PointDataGridT,
-          typename MaskT = typename PointDataGridT::template ValueConverter<bool>::Type>
+          typename MaskT = typename PointDataGridT::template ValueConverter<bool>::Type,
+          typename FilterT = NullFilter>
 inline typename std::enable_if<std::is_same<typename MaskT::ValueType, bool>::value,
     typename MaskT::Ptr>::type
 convertPointsToMask(const PointDataGridT& grid,
-                    const std::vector<Name>& includeGroups = std::vector<Name>(),
-                    const std::vector<Name>& excludeGroups = std::vector<Name>());
+                    const FilterT& filter = NullFilter());
```

Comment:
threaded arguments?

Reply:
These threaded arguments are actually added in the advection PR - https://github.com/dreamworksanimation/openvdb/pull/232. I was just trying to restrict this PR to addressing filtering and counting here.